src/applications/disability-benefits/all-claims/tests/property-names.unit.spec.js

```diff
@@ -0,0 +1,53 @@
+import { expect } from 'chai';
+
+import get from '../../../../platform/utilities/data/get';
+import * as pages from '../pages';
+
+// Some pages have duplicate properties on purpose. These pages add new information to existing
+//  objects and should be ignored.
+const ignoreList = {
+  // propName: ['list', 'of', 'pages'],
+  newDisabilities: ['addDisabilities', 'newDisabilityFollowUp'],
+  serviceInformation: ['federalOrders', 'reservesNationalGuardService'],
+};
+
+describe('Root property names', () => {
+  it('should not be duplicated', () => {
+    const properties = new Map();
+    const duplicatedProperties = {};
+    Object.keys(pages).forEach(pageName => {
+      // Loop through the properties on the schema for a page
+      Object.keys(get(`${pageName}.schema.properties`, pages, {})).forEach(
+        propName => {
+          // Compile a list of duplicate property names
+          if (
+            properties.has(propName) &&
+            !get(propName, ignoreList, []).includes(pageName)
+          ) {
+            if (duplicatedProperties[propName]) {
+              duplicatedProperties[propName].push(pageName);
+            } else {
+              duplicatedProperties[propName] = [
```

Comment:
Took me a while to understand what's going on here, probably because it's still Monday morning 😛
Just so I'm clear, you're pulling the page name of the property in the existing `properties` map, and also the `pageName` of the existing page being iterated, and adding both page names to the array of the new duplicated property, so that both instances of the duplication are identified.

Reply:
Yup! That's exactly it. :smile: